setup.py

```diff
@@ -35,6 +36,48 @@ def run(self):
         os.system('rm -vrf ./build ./dist ./*.pyc ./*.tgz ./*.egg-info')
 
 
+def git_version(version):
+    try:
+        import git
+        repo = git.Repo('.git')
+    except ImportError:
+        print('gitpython not found: Assume release.')
```

Comment:
Nit: Not really assuming a release, instead can have:
gitpython not found: unable to calculate git SHA